notebooks/search/09-semantic-text.ipynb

```diff
@@ -0,0 +1,558 @@
+    "\n",
+    "For this example we'll use the [ELSER service](https://www.elastic.co/guide/en/machine-learning/current/ml-nlp-elser.html), but the inference API also supports [many other inference services](https://www.elastic.co/guide/en/elasticsearch/reference/current/put-inference-api.html#put-inference-api-desc)."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "22fa643780acd44a"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    client.inference.delete_model(inference_id=\"my-elser-endpoint\")\n",
+    "except exceptions.NotFoundError:\n",
+    "    # Inference endpoint does not exist\n",
+    "    pass\n",
+    "\n",
+    "try:\n",
+    "    client.options(\n",
+    "        request_timeout=60, max_retries=3, retry_on_timeout=True\n",
+    "    ).inference.put_model(\n",
+    "        task_type=\"sparse_embedding\",\n",
+    "        inference_id=\"my-elser-endpoint\",\n",
+    "        body={\n",
+    "            \"service\": \"elser\",\n",
+    "            \"service_settings\": {\"num_allocations\": 1, \"num_threads\": 1},\n",
+    "        },\n",
+    "    )\n",
+    "    print(\"Inference endpoint created successfully\")\n",
+    "except exceptions.BadRequestError as e:\n",
+    "    if e.error == \"resource_already_exists_exception\":\n",
+    "        print(\"Inference endpoint created successfully\")\n",
+    "    else:\n",
+    "        raise e"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "8ee2188ea71324f5"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Once the endpoint is created, we must wait until the backing ELSER service is deployed.\n",
+    "This can take a few minutes to complete."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "e94fd66761fd8087"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "inference_endpoint_info = client.inference.get_model(\n",
+    "    inference_id=\"my-elser-endpoint\",\n",
+    ")\n",
+    "model_id = inference_endpoint_info[\"endpoints\"][0][\"service_settings\"][\"model_id\"]\n",
+    "\n",
+    "while True:\n",
+    "    status = client.ml.get_trained_models_stats(\n",
+    "        model_id=model_id,\n",
+    "    )\n",
+    "\n",
+    "    deployment_stats = status[\"trained_model_stats\"][0].get(\"deployment_stats\")\n",
+    "    if deployment_stats is None:\n",
+    "        print(\"ELSER Model is currently being deployed.\")\n",
+    "        continue\n",
+    "\n",
+    "    nodes = deployment_stats.get(\"nodes\")\n",
+    "    if nodes is not None and len(nodes) > 0:\n",
+    "        print(\"ELSER Model has been successfully deployed.\")\n",
+    "        break\n",
+    "    else:\n",
+    "        print(\"ELSER Model is currently being deployed.\")\n",
+    "    time.sleep(5)"
```

Comment:
@joemcelroy I used the [ELSER notebook](https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/search/03-ELSER.ipynb) as a guide for this code and modified it to work with inference endpoints. However, I found two issues when doing so:

- The notebook CI currently doesn't check against a serverless environment
- The [status check used in the ELSER notebook](https://github.com/elastic/elasticsearch-labs/blob/867785b2dcfb3232432c14050d6269fc00116846/notebooks/search/03-ELSER.ipynb#L263) breaks when run against a serverless environment. `trained_model_stats.0.deployment_stats.state` doesn't exist in the serverless response to `client.ml.get_trained_models_stats`. Instead, I checked if the `nodes` list is populated, which seems to work in both hosted & serverless environments.